timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -234,13 +234,6 @@ def _get_hpo_backend(self):
 
     def _deferred_init_params_aux(self, dataset: TimeSeriesDataFrame) -> None:
         """Update GluonTS specific parameters with information available only at training time."""
-        self.freq = dataset.freq or self.freq
-        if not self.freq:
```

Comment:
1/ Why remove `self.freq = dataset.freq or self.freq`? We never depend on deferred initialization for freq in any other model? Is freq always set for all models by the trainer?
2/ Why remove this assertion? maybe `assert self.freq`?

Reply:
1. No other model currently depends on deferred initialization of `freq` (both local & MLForecast models assume that `freq` is provided at creation time); trainer always sets `freq` when creating models (https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py#L576)
2. Assuming you are referring to the assertion inside `SimpleGluonTSDataset.__init__`: after pandas 2.2 upgrade, we will anyway need to remove this part completely and always feed a dummy frequency `W` to GluonTS. So the assertion would be removed anyway after #4074 is merged.